modules/pubwiseAnalyticsAdapter.js

```diff
@@ -54,72 +73,268 @@ function enrichWithUTM(dataBag) {
+}
+
+function sessionExpired() {
+  let sessLastTime = localStorage.getItem(localStorageSessTimeoutName());
+  return (Date.now() - parseInt(sessLastTime)) > sessTimeout;
+}
+
+function flushEvents() {
+  if (pwEvents.length > 0) {
+    let dataBag = {metaData: metaData, eventList: pwEvents.splice(0)}; // put all the events together with the metadata and send
+    ajax(configOptions.endpoint, (result) => pwInfo(`Result`, result), JSON.stringify(dataBag));
+  }
+}
+
+function isIngestedEvent(eventType) {
+  const ingested = [
+    CONSTANTS.EVENTS.AUCTION_INIT,
+    CONSTANTS.EVENTS.BID_REQUESTED,
+    CONSTANTS.EVENTS.BID_RESPONSE,
+    CONSTANTS.EVENTS.BID_WON,
+    CONSTANTS.EVENTS.BID_TIMEOUT,
+    CONSTANTS.EVENTS.AD_RENDER_FAILED,
+    CONSTANTS.EVENTS.TCF2_ENFORCEMENT
+  ];
+  return ingested.indexOf(eventType) !== -1;
+}
+
+function markEnabled() {
+  pwInfo(`Enabled`, configOptions);
+  pwAnalyticsEnabled = true;
+  setInterval(flushEvents, 100);
+}
+
+function pwInfo(info, context) {
+  utils.logInfo(`${analyticsName} ` + info, context);
+}
+
+/*
+  // unused currently
+  function filterNoBid(data) {
+    let newNoBidData = {};
 
-  // put the typical items in the data bag
-  let dataBag = {
-    eventType: eventType,
-    args: data,
-    target_site: configOptions.site,
-    pubwiseSchema: pubwiseSchema,
-    debug: configOptions.debug ? 1 : 0,
-  };
+    newNoBidData.auctionId = data.auctionId;
+    newNoBidData.bidId = data.bidId;
+    newNoBidData.bidderRequestId = data.bidderRequestId;
+    newNoBidData.transactionId = data.transactionId;
 
-  dataBag = enrichWithMetrics(dataBag);
-  // for certain events, track additional info
-  if (eventType == CONSTANTS.EVENTS.AUCTION_INIT) {
-    dataBag = enrichWithUTM(dataBag);
+    return newNoBidData;
   }
+*/
 
-  ajax(configOptions.endpoint, (result) => utils.logInfo(`${analyticsName}Result`, result), JSON.stringify(dataBag));
+function filterBidResponse(data) {
+  let modified = Object.assign({}, data);
+  // clean up some properties we don't track in public version
+  if (typeof modified.ad !== 'undefined') {
+    modified.ad = '';
+  }
+  if (typeof modified.adUrl !== 'undefined') {
+    modified.adUrl = '';
+  }
+  if (typeof modified.adserverTargeting !== 'undefined') {
+    modified.adserverTargeting = '';
+  }
+  if (typeof modified.ts !== 'undefined') {
+    modified.ts = '';
+  }
+  // clean up a property to make simpler
+  if (typeof modified.statusMessage !== 'undefined' && modified.statusMessage === 'Bid returned empty or error response') {
+    modified.statusMessage = 'eoe';
+  }
+  modified.auctionEnded = auctionEnded;
+  return modified;
 }
 
-let pubwiseAnalytics = Object.assign(adapter(
-  {
-    defaultUrl,
-    analyticsType
-  }),
-{
+function filterAuctionInit(data) {
+  let modified = Object.assign({}, data);
+
+  modified.refererInfo = {};
+  // handle clean referrer, we only need one
+  if (typeof modified.bidderRequests !== 'undefined') {
```

Comment:
again, consider the more succint syntax `if (modified.bidderRequests && modified.bidderRequests[0] && ... )`

Reply:
We don't rely on truthiness for data not under our control. Paranoid? Maybe? Unecessary? Maybe. We write in 6 different languages and would like to reliably get the meaning we want. Our desire is to determine if the values is, but often is not, defined and underlying behavior in various languages around what happens when the value is defined but "trueish" is variable. So, we make explicit statements when the value comes from somewhere else.